common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -791,8 +812,9 @@ private synchronized Object progressiveListeners() {
             }
 
             return copy;
-        } else if (listeners instanceof GenericProgressiveFutureListener) {
-            return listeners;
+        } else if (listener instanceof GenericProgressiveFutureListener) {
+            // TODO can we save the instanceof and just makes use of the progressiveSize?
```

Comment:
@normanmaurer ^^

Reply:
Progressive promises and futures are not used very often, so fine to not optimize IMO.